### PR TITLE
fix the testing documentation

### PR DESCRIPTION
**Pull request by VitalinaVZdrobau**
Fixing the testing documentation
Added 'TestCafe - Functional Testing' page
Added 'Running API tests with Karate' steps for java API projects
Deleted the 'Performance testing' page from .Net (as the info is duplicated on the 'Performance Testing - Gatling') page
Deleted the 'Testing the API net core' from the 'sidebars.js' file

### CODE REVIEW

File: docs/testing/axe_automated_accessibility_testing.md [hunk 1]
```diff
@@ -12,8 +12,8 @@ expect to support WCAG 2.1 Level AA ["wcag21aa"] at a minimum.
 
 aXe tests are performed on two levels:
 
-1. first by rendering the React component and testing with Jest.
-2. second with Next rendering the entire page and testing with Cypress.
+1. By rendering the React component and testing with Jest.
```
Comment (jpbsheppard) [inline docs/testing/axe_automated_accessibility_testing.md:15]: Rendering the React component and testing with Jest.

File: docs/testing/axe_automated_accessibility_testing.md [hunk 2]
```diff
@@ -12,8 +12,8 @@ expect to support WCAG 2.1 Level AA ["wcag21aa"] at a minimum.
...
+2. With Next rendering the entire page and testing with Cypress.
```
Comment (jpbsheppard) [inline docs/testing/axe_automated_accessibility_testing.md:16]: Rendering the entire page and testing with Cypress.

File: docs/testing/cli_testing_frameworks.md [hunk 3]
```diff
@@ -4,22 +4,22 @@ title: Stacks Testing Frameworks
 sidebar_label: Stacks Testing Frameworks
 ---
 
-Building a testing frameworks from scratch requires a lot of boilerplate code. It is also hard to know how to get started without examples.
+Building a testing framework from scratch requires a lot of boilerplate code. It is also hard to know how to get started without examples.
```
Comment (jpbsheppard) [inline docs/testing/cli_testing_frameworks.md:7]: Building testing frameworks from scratch requires a lot of boilerplate code. It is also hard to know how to get started without examples.

File: docs/testing/cli_testing_frameworks.md [hunk 4]
```diff
@@ -4,22 +4,22 @@ title: Stacks Testing Frameworks
...
 
 As part of Amido Stacks projects, we have developed a range of Javascript Front End focused, opinionated, boilerplate testing frameworks that can be bootstrapped and tailored to your current project with just one command.
 
 ## Bootstrap your framework
 
-The best way to get started, is bootstrap the framework using the scaffolding-cli from Amido Stacks. By using it in interactive mode, you can simply answer 5 questions to get a bespoke framework, ready to go with an example test.
+The best way to get started is to bootstrap the framework using the scaffolding-cli from Amido Stacks. By using it in the interactive mode, you can simply answer 5 questions to get a bespoke framework, ready to go with an example test.
```
Comment (jpbsheppard) [inline docs/testing/cli_testing_frameworks.md:13]: The best way to get started is to bootstrap the framework using the scaffolding-cli from Amido Stacks. By using it in interactive mode, you can simply answer 5 questions to get a bespoke framework ready to go with an example test.

File: docs/testing/cypress_functional_testing.md [hunk 5]
```diff
@@ -10,10 +10,10 @@ We are using [Cypress](https://docs.cypress.io/) for functional testing as much
 as possible. The key features of Cypress:
 
 - Integration testing with API testing support (Node.js)
-- Easy debugability (DOM snapshotting)
+- Easy debuggability (DOM snapshotting)
```
Comment (jpbsheppard) [inline docs/testing/cypress_functional_testing.md:13]: Easy to debug

File: docs/testing/dotnet_testing_selenium.md [hunk 6]
```diff
@@ -73,6 +73,8 @@ In root (*.sln) directory, build the solution using: ) the solution:
 dotnet test
 ```
 
+As a result of the test execution - the 'BDDfy.html' test report will be automatically generated.
```
Comment (jpbsheppard) [inline docs/testing/dotnet_testing_selenium.md:76]: As a result of test execution - a 'BDDfy.html' test report will be automatically generated.

File: docs/testing/java_testing_serenity_bdd.md [hunk 7]
```diff
@@ -111,7 +118,7 @@ mvn serenity:aggregate
 Using multiple forks can be a good alternative to running all tests in a single JVM, and can reduce the risk of certain types of errors.
 
 :::info
-Choosing the right forking strategy and parallel execution settings can have a substantial impact on the memory requirements and the execution time of the build system.
+Choosing the right forking strategy and parallel execution settings can have a substantial impact on the memory requirements, and the execution time of the build system.
```
Comment (jpbsheppard) [inline docs/testing/java_testing_serenity_bdd.md:121]: , and is a grammatical error. Revert this change

File: docs/testing/java_testing_serenity_bdd.md [hunk 8]
```diff
@@ -120,7 +127,7 @@ This creates better separation between tests, which can improve their reliabilit
 
 Maven is set to spawn new processes by the forkCount configuration element, as shown here below.
 
-This can either be a number (the maximum number of forks) or a multiplier (the number of forks per CPU). The current configuration uses the value of "2", which means 2 forked processes per CPU:
+This can either be a number (the maximum number of forks), or a multiplier (the number of forks per CPU). The current configuration uses the value of "2", which means 2 forked processes per CPU:
```
Comment (jpbsheppard) [inline docs/testing/java_testing_serenity_bdd.md:130]: Revert

File: docs/testing/testcafe_functional_testing.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
+---
+id: testcafe_functional_testing
+title: Functional Testing with TestCafe
+sidebar_label: TestCafe - Functional Testing
+---
+
+## TestCafe: Functional tests
+
```
Comment (jpbsheppard) [inline docs/testing/testcafe_functional_testing.md:8]: Add a section here about what TestCafe is, what it is generally used for and why it is beneficial as well as a link to TestCafe itself.

File: docs/testing/testcafe_functional_testing.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+Templated sample TestCafe automation framework for PROJECT_NAME, bootstrapped using `@amidostacks/scaffolding-cli`.
+
+## Getting started
+
+Ensure that [Node.js](https://nodejs.org/) and [npm](https://www.npmjs.com/) are installed with the versions specified in [package.json](./package.json).
```
Comment (jpbsheppard) [inline docs/testing/testcafe_functional_testing.md:13]: Add this to a Prerequisites section

File: docs/testing/testcafe_functional_testing.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+### End to End (E2E) Tests
+
+```bash
+npm run test:e2e
+```
+
+This should launch [TestCafe](https://devexpress.github.io/testcafe/documentation/getting-started/) and run the tests in the browsers specified in [.testcaferc.json](./.testcaferc.json) in the headless mode.
+
+### Accessibility Tests with aXe
+
```
Comment (jpbsheppard) [inline docs/testing/testcafe_functional_testing.md:30]: This requires some explanation as you have gone from Functional Testing with TestCafe to Accessibility testing with aXe.  A section renamed as Adding accessibility Tests with aXe to TestCafe perhaps.  You should also move this section until after you have fully gone through functional testing, so further down the page

File: docs/testing/testcafe_functional_testing.md [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+npm run test:axe
+```
+
+This will launch [TestCafe](https://devexpress.github.io/testcafe/documentation/getting-started/) and run the tests in the browsers specified in [.testcaferc.json](./.testcaferc.json) in headless mode.
+
+This will validated using axe-core, from the Deque family of products: [aXe](https://www.deque.com/axe/). When developing, we expect to support WCAG 2.1 Level AA ["wcag21aa"] at a minimum. The additional configuration can be changed by defining the [context and options](https://www.npmjs.com/package/axe-testcafe#axe-options).
+
+## Browser configuration
+
+We assume that most users will have Chrome installed on their operating system. If this is not the case, the base configuration file
```
Comment (jpbsheppard) [inline docs/testing/testcafe_functional_testing.md:41]: Never assume, if Chrome is a dependency then mention that in dependencies.

Comment (jpbsheppard) [inline docs/testing/testcafe_functional_testing.md:41]: Finish this sentence: If this is not the case, the base configuration file

File: docs/testing/testcafe_functional_testing.md [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+``` bash
+export NODE_ENV=development 
+export PORT=3000
+export APP_BASE_URL=APP_BASE_URL_VALUE
+export MENU_API_URL=MENU_API_URL_VALUE
+export APP_BASE_PATH=""
+```
+
+## Running tests in Docker
+
```
Comment (jpbsheppard) [inline docs/testing/testcafe_functional_testing.md:81]: Given the instructions here there is a lot of dependencies to get this to run.  I would add in links to docker and how to set this up here otherwise you will have no context

File: docs/testing/testcafe_functional_testing.md [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+docker pull testcafe/testcafe:latest
+```
+
+```bash
+docker run -e APP_BASE_URL=$APP_BASE_URL -e APP_BASE_PATH=$APP_BASE_PATH -e MENU_API_URL=$MENU_API_URL -e NODE_ENV=$NODE_ENV -it -v $(pwd):/tests testcafe/testcafe chromium /tests/**/*.test.cf.ts
+```
+
+### Running in CI/CD
+
+Currently, we are supporting running Azure Pipelines. Please refer to the [test-functional-testcafe.yml](https://github.com/amido/stacks-pipeline-templates/blob/feature/cycle2/azDevOps/azure/templates/v2/steps/test-functional-testcafe.yml) for information on the step.
```
Comment (jpbsheppard) [inline docs/testing/testcafe_functional_testing.md:102]: Currently, running in CI/CD is supported only in Azure Pipelines

File: docs/testing/testcafe_functional_testing.md [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+
+We run the TestCafe functional tests after successful deployment of the webapp to the cluster.
```
Comment (jpbsheppard) [inline docs/testing/testcafe_functional_testing.md:104]: The TestCafe functional tests are configured to run after a successful deployment of the webapp to the cluster.

File: docs/testing/testcafe_functional_testing.md [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+
+The `publishArtifact` task ensures the screenshots are captured if an applicable test fails. These are available as artifacts in the pipeline.
+
+## Running with Lambdatest
+
+[Lambdatest](https://www.lambdatest.com) is a Cross Browser Testing Cloud platform, leveraging the running on 2000+ Real Browsers and Operating Systems Online.
```
Comment (jpbsheppard) [inline docs/testing/testcafe_functional_testing.md:110]: Think you need to expand this, Im assuming this at lest requires some form of account with Lambdatest and that should be detailed here

File: docs/workloads/azure/backend/java/testing_java.md [hunk 17]
```diff
@@ -111,20 +111,20 @@ To generate the Serenity report:
...
+  mvnw.cmd clean verify
   ```
 
   </TabItem>
  </Tabs>
 
-The report can be viewed under `<PROJECT-NAME>/api-tests/target/site/serenity/index.html`.
+The report can be viewed under `./api-tests/target/site/serenity/index.html`.
 
 #### Code coverage
```
Comment (jpbsheppard) [inline docs/workloads/azure/backend/java/testing_java.md:130]: Little passage here about what tool is being used for code coverage and how it works

File: docs/workloads/azure/backend/java/testing_java.md [hunk 18]
```diff
@@ -141,20 +141,20 @@ Run the following command:
...
+  mvnw.cmd jacoco:report
   ```
 
   </TabItem>
  </Tabs>
 
-The generated report can be viewed under `<PROJECT-NAME>/target/site/jacoco/index.html`.
+The generated report can be viewed under `./target/site/jacoco/index.html`.
 
 #### Dependency checker report
```
Comment (jpbsheppard) [inline docs/workloads/azure/backend/java/testing_java.md:160]: Same here, little passage about what we are using and how it works

File: docs/workloads/azure/backend/java/testing_java.md [hunk 19]
```diff
@@ -169,20 +169,20 @@ The generated report can be viewed under `<PROJECT-NAME>/target/site/jacoco/inde
...
+  mvnw.cmd clean install -Powasp-dependency-check
   ```
 
   </TabItem>
  </Tabs>
 
-The generated report can be viewed under `<PROJECT-NAME>/target/dependency-check.html`.
+The generated report can be viewed under `./target/dependency-check.html`.
 
 #### Mutation testing
```
Comment (jpbsheppard) [inline docs/workloads/azure/backend/java/testing_java.md:188]: This requires clarification about what it is and how its being done.  This appears to come out of nowhere and is not present in any of the other workloads.

Comment (jpbsheppard) [reply docs/workloads/azure/backend/java/testing_java.md:188]: If this is an optional thing that can be done then it should move into a section below as an option that can be used as it strictly is not mandatory.

File: docs/workloads/azure/backend/java/testing_java.md [hunk 20]
```diff
@@ -297,3 +302,62 @@ and to run the Functional tests independently, use
 
   </TabItem>
  </Tabs>
+ </details>
+ 
+ <br />
+ 
+<details>
+<summary> Running API tests with Karate </summary>
```
Comment (jpbsheppard) [inline docs/workloads/azure/backend/java/testing_java.md:310]: Think this needs to be its own file as realistically you wouldn't run Karate Tests if you were using Serenity.  The Karate test framework and Serenity should be completely standalone

**Review by corradot12 — APPROVED** [review]

File: docs/workloads/azure/backend/java_cqrs/testing_java_cqrs.md [hunk 21]
```diff
@@ -70,7 +70,7 @@ and the equivalent for running the Integration tests.
 #### Test reports
```
Comment (jpbsheppard) [inline docs/workloads/azure/backend/java_cqrs/testing_java_cqrs.md:70]: The same changes that are to be made in the above file should be made in this one

**Review by jpbsheppard — CHANGES_REQUESTED** [review]

File: docs/testing/java_testing_serenity_bdd.md [hunk 22]
```diff
@@ -122,7 +129,7 @@ This creates better separation between tests, which can improve their reliabilit
 
 Maven is set to spawn new processes by the forkCount configuration element, as shown here below.
 
-This can either be a number (the maximum number of forks) or a multiplier (the number of forks per CPU). The current configuration uses the value of "2", which means 2 forked processes per CPU:
+his can either be a number (the maximum number of forks) or a multiplier (the number of forks per CPU). The current configuration uses the value of "2", which means 2 forked processes per CPU:
```
Comment (jpbsheppard) [inline docs/testing/java_testing_serenity_bdd.md:132]: little typo

**Review by jpbsheppard — CHANGES_REQUESTED** [review]
One little typo

File: docs/testing/java_testing_serenity_bdd.md (hunk 22 above)
Comment (VitalinaVZdrobau) [reply docs/testing/java_testing_serenity_bdd.md]: fixed

**Review by jpbsheppard — APPROVED** [review]
